datacodec/errors.go

```diff
@@ -219,6 +227,10 @@ func collectionSizeNegative(size int) error {
 	return fmt.Errorf("expected collection size >= 0, got: %d", size)
 }
 
+func collectionElementNil() error {
+	return fmt.Errorf("nil is not supported inside collections")
+}
```

Comment:
oh we don't have proper error types in this library? that sucks, we need to do this at some point but for now it's ok